src/plugins/gui/providers/paybisProvider.ts

```diff
@@ -757,17 +754,24 @@ export const paybisProvider: FiatProviderFactory = {
                 lstrings.fiat_plugin_cannot_continue_camera_permission
               )
             }
-            const receiveAddress = await coreWallet.getReceiveAddress({
+            const [receiveAddress] = await coreWallet.getAddresses({
               tokenId: null
             })
+            assert(receiveAddress != null, 'Paybis: Missing receive address')
+            const receiveSegwitAddress = (
+              receiveAddress as {
+                segwitAddress?: string
+              }
```

Comment:
Thanks for flagging. This is safe: coreWallet.getAddresses({ tokenId }) always returns at least one entry. In the fallback path we always include a publicAddress, and if a segwit address exists it is unshifted to the front, so index 0 is valid. The first entry publicAddress will already be the segwit address when present. Added a defensive check where appropriate.